Algorithms/Linked List/linkedlist.java

```diff
@@ -0,0 +1,189 @@
+		this.head = null;
+		this.tail = null;
+		this.size = 0;
+	}
+
+	// this function returns true if the linked list is empty
+	public boolean isEmpty() {
+		return this.size == 0;
+	}
+
+	// this function returns the size of the linked list
+	public int size() {
+		return this.size;
+	}
+
+	// this function adds a node in the beginning of the linked list
+	public void addFirst(int data) {
+
+		// declaring node to be added
+		Node node = new Node(data, this.head);
+
+		// this makes head of the linked list point to the new node
+		this.head = node;
+
+		// if the linked list is empty, we change the tail pointer as well
+		if (this.isEmpty()) {
+			this.tail = node;
+		}
+
+		// and increase the size of the linked list
+		this.size++;
+	}
+
+	// this function adds a node in the end of the linked list
+	public void addLast(int data) {
+
+		// declaring node to be added
+		Node node = new Node(data, null);
+
+		// if the linked list is empty, we change head and tail both
+		if (this.isEmpty()) {
+			this.head = node;
+			this.tail = node;
+		} else {
+			// if linked list is not empty, we change only tail pointer
+			this.tail.next = node;
+			this.tail = node;
+		}
+
+		// and increase the size of the linked list
+		this.size++;
+	}
+
+	// This function displays the linked list
+	public void display() {
+
+		// we start traversing the list from head
+		Node temp = this.head;
+
+		// and until this temp node becomes null
+		while (temp != null) {
+
+			// we print it's data
+			System.out.print(temp.data + "=>");
+
+			// and move the temp pointer forward
+			temp = temp.next;
+		}
+
+		System.out.println("END");
+
+	}
+
+	public boolean hasCycle() {
+		return this.hasCycle(this.head);
+	}
+
+	// Floyd�s Cycle-Finding Algorithm
+	// This function returns true if there is a cycle in the linked list
+	private boolean hasCycle(Node head) {
+
+		// if there are no nodes in a linked list, there can't be any cycle present
+		if (head == null) {
+			return false;
+		}
+
+		// we take two pointers slow and fast and start traversing the list, initially
+		// both point at head
+		Node slow = head;
+		Node fast = head;
+
+		// If there is no cycle present, then slow or fast or fast.next will become null
+		// at some point of time
+		while (slow != null && fast != null && fast.next != null) {
+
+			// in every iteration, slow moves ahead by one node
+			slow = slow.next;
+
+			// and fast moves ahead by two nodes
+			fast = fast.next.next;
+
+			// if slow and fast reach the same node, then there is a cycle present, so we
+			// return true
+			if (slow == fast) {
+				return true;
+			}
+		}
+
+		// if there is no cycle present and we have traversed the whole linked list, we
+		// return false
+		return false;
+	}
+
+	public static void main(String[] args) {
+		// TODO Auto-generated method stub
+
+		// declaring a linked list
+		linkedlist list = new linkedlist();
+
+		// adding elements in the list
+		list.addFirst(10);
+		list.addLast(20);
```

Comment:
@Aashu24 Your code has an awesome explanation as well as you have written it in a good way. Just prompt user for input data in linked list and you are good to go.